modules/ding_webtrekk/ding_webtrekk.js

```diff
@@ -0,0 +1,131 @@
+      });
+
+      // Track autocomplete selections.
+      $('.form-item-search-block-form .form-autocomplete', context)
+        .once('ding-webtrekk')
+        .on('autocompleteSelect', function(e, selected) {
+          if (($(selected).text())) {
+            wts.push(['send', 'click', {
+              linkId: 'Autocomplete søgning clicks',
+              customClickParameter: { 54: $(selected).text() }
+            }]);
+          }
+      });
+
+      // Track ding loan renew selected events.
+      //
+      // The renew all button is attached completely on the server, but for
+      // renew selected, we need information about selected loans in the UI,
+      // before we send event.
+      $('.ding-webtrekk-event-renew-selected', context)
+        .once('ding-webtrekk')
+        .click(function(e) {
+          var selectedMaterials = [];
+          $('.material-item input[type=checkbox]:checked').each(function() {
+            // We have collected the material ids (pids) in our event data
+            // attribute for each select box.
+            selectedMaterials.push($(this).data('ding-webtrekk-event'));
+          });
+          wts.push(['send', 'click', {
+            linkId: 'Forny valgte materialer',
+            customClickParameter: { 55: selectedMaterials.join(';') }
+          }]);
+      });
+
+      // Special handling for ding_carousel.
+      //
+      // The primary reason for this is that we need the title of the carousel
+      // to pass as value in event and URL-parameter. For tabbed carousels the
+      // title of the individuals carousels in the tabs are set in the
+      // data-title attribute of the carousels. For single carousels the title
+      // is not avialable here, but instead it's controlled by the configuration
+      // on the panel pane. There are no panel hooks where we can get the title
+      // and easily change the content tree to add event and URL-parameters to
+      // elements (content is already rendered when hook is run).
+      $('.ding-carousel').each(function() {
+        var carouselTitle = false;
+        var key = 'u_navigatedby';
+
+        // We need a title to send as value in the 'u_navigatedby' parameter, so
+        // if we can't find a carousel title we will just have to fallback to
+        // this generic one.
+        carouselTitle = 'unknown_carousel';
+
+        if ($(this).data('title')) {
+          carouselTitle = $(this).data('title');
+        }
+        else if ($(this).parent().siblings('.pane-title').length > 0) {
+          carouselTitle = $(this).parent().siblings('.pane-title').html();
+        }
+
+        // Add tracking URL-pararmeters to items in the carousel.
+        $('.ding-carousel-item .ting-object a[href^="/ting/object/"]', this).once('ding-webtrekk', function() {
+          $(this).attr('href', appendQueryParameter($(this).attr('href'), key, carouselTitle));
+        });
+
+        $('.slick-arrow', this).once('ding-webtrekk').click(function() {
```

Comment:
Is there a reason why you do not use [the events provided by Slick](http://kenwheeler.github.io/slick/) for this?

Reply:
I wasn't aware of the events to be honest. Which event do you suggest we use? After looking at the available events it seems to be more complicated to determine if it was prev or next that was clicked.

Reply:
I looked a bit further into this. There doesn't seem to be any events for the arrow-button and therefore the easiest approach seems to use custom click handlers as i done now in the PR.

Reply:
There is a swipe event, which we could use to send event, when mobile users swipes the carousel or it's dragged. I believe the business hasn't thought about this when choosing what to track and I have asked them about it.

Reply:
I asked the business and there was no interest in tracking swipes/drags on carousel.